File: Web-Api/BLL/Services/PickService.cs
using Pickfc.BLL.Interfaces;
using Pickfc.DAL.Infrastructure;
using Pickfc.Model.Context;
using Pickfc.Model.Entities;
using Pickfc.DAL.Interfaces;

namespace Pickfc.BLL.Services
{
    public class PickService : IPickService
    {
        private readonly IPickRepository pickRepository;
        private readonly WorkUnit<PickfcContext> workUnit;

        public PickService(IPickRepository pickRepository, WorkUnit<PickfcContext> workUnit) {
            this.pickRepository = pickRepository;
            this.workUnit = workUnit;
        }

        public Pick Pick(int id) {
            return pickRepository.SingleOrDefault(x => x.ID == id);
        }

        public IEnumerable<Pick> Picks(int roundid)
        {
            return pickRepository.GetMany(x => x.RoundID == roundid);
        }

        public Pick PlayerPick(int playerid, int gameid, int roundid) 
        {
            return pickRepository.SingleOrDefault(x => x.PlayerID == playerid && x.GameID == gameid && x.RoundID == roundid);
        }

        public IEnumerable<Pick> PlayerPicks(int playerid, int gameid) 
        {
            return pickRepository.GetMany(x => x.PlayerID == playerid && x.GameID == gameid).OrderByDescending(x => x.Timestamped);
        }

        public Pick Add(Pick pick) 
        { 
            pick.Timestamped = DateTime.Now;
            pick.Time = DateTime.Now;
            pickRepository.Add(pick);
            workUnit.Commit();
            return pick;
        }

        public void Edit(Pick pick) 
        { 
            pickRepository.Update(pick);
            workUnit.Commit();
        }

        public void Delete(Pick pick)
        {
            if (pick != null) 
            { 
                pickRepository.Delete(pick);
                workUnit.Commit();
            }
        }

        public bool PickValid(int playerid, int teamid) 
        {
            return pickRepository.Any(x => x.PlayerID == playerid && x.TeamID != teamid);
        }

        public bool PickExist(int playerid, int gameid, int roundid) {
            return pickRepository.Any(x => x.PlayerID == playerid && x.GameID == gameid && x.RoundID == roundid);
        }
    }
}

File: Client-App/app/_service/pick.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Pick } from '../_model/pick';
import { Player } from '../_model/player';

@Injectable({
  providedIn: 'root'
})
export class PickService {

  private readonly url: string = environment.baseUrl;

  constructor(private http: HttpClient) { }

  public pick(id: number): Observable<Pick> {
    return this.http.get<Pick>(`${this.url}api/Pick/Pick?id=${id}`);
  }

  public playerPick(player: Player, roundid: number): Observable<Pick> {
    return this.http.get<Pick>(`${this.url}api/Pick/Pick?playerid=${player.id}&gameid=${player.gameID}&roundid=${roundid}`);
  }

  public playerPicks(player: Player): Observable<Pick[]> {
    return this.http.get<Pick[]>(`${this.url}api/Pick/PlayerPicks?playerid=${player.id}&gameid=${player.gameID}`);
  }

  public picks(roundid: number): Observable<Pick[]> {
    return this.http.get<Pick[]>(`${this.url}api/Pick/Picks?roundid=${roundid}`);
  }

  public pickValid(playerid: number, roundid: number): Observable<boolean> {
    return this.http.get<boolean>(`${this.url}api/Pick/PickValid?playerid=${playerid}&roundid=${roundid}`);
  }

  public pickExist(pick: Pick): Observable<boolean> {
    return this.http.get<boolean>(`${this.url}api/Pick/PickExist?playerid=${pick.playerID}&gameid=${pick.gameID}&roundid=${pick.roundID}`);
  }

  public addedit(pick: Pick): Observable<Pick> {
    return pick.id === 0 ? this.http.post<Pick>(`${this.url}api/Pick/Add`, pick) : this.http.put<Pick>(`${this.url}api/Pick/Edit`, pick);
  }

  public delete(id: number): Observable<Pick> {
    return this.http.delete<Pick>(`${this.url}api/Pick/Delete?id=${id}`)
  }

}
